pkg/controller/job/job_controller_test.go

```diff
@@ -1838,29 +1845,53 @@ func TestSyncPastDeadlineJobFinished(t *testing.T) {
 
 	go manager.Run(ctx, 1)
 
-	job := newJob(1, 1, 6, batch.NonIndexedCompletion)
-	activeDeadlineSeconds := int64(1)
-	job.Spec.ActiveDeadlineSeconds = &activeDeadlineSeconds
-	_, err := clientset.BatchV1().Jobs(job.GetNamespace()).Create(ctx, job, metav1.CreateOptions{})
-	if err != nil {
-		t.Errorf("Could not create Job: %v", err)
-	}
-	if err := sharedInformerFactory.Batch().V1().Jobs().Informer().GetIndexer().Add(job); err != nil {
-		t.Fatalf("Failed to insert job in index: %v", err)
+	tests := []struct {
+		name      string
+		startTime *metav1.Time
```

Comment:
make it a boolean instead and create the time within the implementation. Otherwise `time.Now` could be too old for it to be a useful test.

But reiterating: we need to make the test have a fake clock as a follow up.